src/mqueryfront/src/QueryPage.js

```diff
@@ -79,8 +104,8 @@ class QueryPage extends Component {
             queryError: null,
             queryPlan: null,
             qhash: newQhash,
-            matches: [],
-            job: [],
+            matches: null,
```

Comment:
Why should matches be null instead of `[]`? Matches represent a `list of matches`, so empty list sounds like a good default. But if it works correctly (or if there's something I don't see) i think it's OK too.

Reply:
An empty list is a good default for the match list. But the code used two defaults: [] and null. I unified the code to consistently use null everywhere.